dev/com.ibm.ws.logging_2_fat/fat/src/com/ibm/ws/logging/fat/MessageTraceFileNameTimedRolloverTest.java

```diff
@@ -0,0 +1,427 @@
+        if (waitMs > 0)
+            Thread.sleep(waitMs);
+
+        int rolled = waitForRolledFiles("trace", 5000);
+        assertTrue("trace_* rolled files should exist after update to trace.log", rolled > 0);
+
+        System.out.println("=== TEST 3: COMPLETE ===");
+    }
+
+    /**
+     * Test 4: With TimedLogRollover, dynamically update traceFileName to test.log,
+     * verify test.log CREATES and ROLLS, and verify trace_* does NOT roll.
+     */
+    @Test
+    public void testDynamicTraceFileNameToCustomFile() throws Exception {
+        System.out.println("=== TEST 4: Requirements Check ===");
+        System.out.println("✓ Dynamically update traceFileName to test.log");
+        System.out.println("✓ Verify test.log gets rolled over");
+        System.out.println("✓ Verify trace.log does NOT roll over");
+
+        server.startServer();
+        server.waitForStringInLog("CWWKF0011I");
+
+        // Sanity: trace.log should not be there at start of this test run
+        boolean initialTraceLog = server.fileExistsInLibertyServerRoot("logs/trace.log");
+        System.out.println("Initial trace.log exists: " + initialTraceLog);
+
+        // Update to test.log
+        server.setMarkToEndOfLog();
+        server.setServerConfigurationFile("server_test_log_chatty.xml");
+        assertNotNull("Config update should complete", server.waitForConfigUpdateInLogUsingMark(null));
+
+        // test.log must be created shortly after update
+        assertTrue("test.log should be created after dynamic update",
+                   waitForFileExists("logs/test.log", 15000));
+
+        // On next boundary: test_* should roll, trace_* must NOT roll
+        Calendar next = getNextRolloverTime(0, 1);
+        long waitMs = next.getTimeInMillis() - System.currentTimeMillis() + 5000;
+        if (waitMs > 0)
+            Thread.sleep(waitMs);
+
+        int testRolled = waitForRolledFiles("test", 5000);
+        int traceRolled = waitForRolledFiles("trace", 2000);
+
+        assertTrue("test_* rolled files should exist", testRolled > 0);
+        assertEquals("trace_* must NOT roll when traceFileName points to test.log", 0, traceRolled);
+
+        System.out.println("=== TEST 4: COMPLETE ===");
+    }
+
+    /**
+     * Test 5: Repeat scenario 4 using messageFileName, instead.
+     */
+    @Test
+    public void testDynamicMessageFileNameToCustomFile() throws Exception {
+        System.out.println("=== TEST 5: Requirements Check ===");
+        System.out.println("✓ Requirement: Dynamically update messageFileName to custom_message.log");
+        System.out.println("✓ Requirement: Verify custom_message.log gets rolled over");
+
+        server.startServer();
+        server.waitForStringInLog("CWWKF0011I");
+
+        server.setMarkToEndOfLog();
+        server.setServerConfigurationFile("server_custom_message.xml");
+        System.out.println("Updating to server_custom_message.xml...");
+
+        server.waitForStringInLog("CWWKG0017I");
```

Comment:
Should use this `assertNotNull("Config update should complete", server.waitForConfigUpdateInLogUsingMark(null));` instead.